### PR TITLE
link traces from repo_update

### PR DESCRIPTION
**Pull request by LMG**
Link to the parent which is in the parent josh process

### CODE REVIEW

**Comment by LMG** [comment]
So I suppose this does improve things a bit, although it might look a bit confusing since the repo_update_fn loses the proper parent. On the plus side, it looks like Jaeger most of the time manages to keep it together with the corresponding http_request

File: josh-proxy/src/bin/josh-proxy.rs [hunk 1]
```diff
@@ -615,6 +627,15 @@ async fn call_service(
...
     }
 
+    let span = tracing::span!(tracing::Level::TRACE, "hyper_cgi");
+    let _enter = span.enter();
+    let mut context_propagator = HashMap::<String, String>::default();
+    let context = span.context();
+    global::get_text_map_propagator(|propagator| {
+        propagator.inject_context(&context, &mut context_propagator);
+    });
+    tracing::warn!("debug propagator: {:?}", context_propagator);
```
Comment (vlad-ivanov-name) [inline josh-proxy/src/bin/josh-proxy.rs:637]: leftover debug?

File: josh-proxy/src/bin/josh-proxy.rs [hunk 2]
```diff
@@ -636,10 +658,8 @@ async fn call_service(
     cmd.env("JOSH_REPO_UPDATE", serde_json::to_string(&repo_update)?);
     cmd.env("PATH_INFO", parsed_url.pathinfo.clone());
 
-    let cgires = hyper_cgi::do_cgi(req, cmd)
-        .instrument(tracing::span!(tracing::Level::TRACE, "git http-backend"))
-        .await
-        .0;
+    let git_span = tracing::span!(tracing::Level::TRACE, "git http backend");
+    let cgires = hyper_cgi::do_cgi(req, cmd).instrument(git_span).await.0;
```
Comment (vlad-ivanov-name) [inline josh-proxy/src/bin/josh-proxy.rs:662]: should result of git execution go into the trace as well?

File: josh-proxy/src/bin/josh-proxy.rs [hunk 3]
```diff
@@ -615,6 +627,15 @@ async fn call_service(
         }
     }
 
+    let span = tracing::span!(tracing::Level::TRACE, "hyper_cgi");
+    let _enter = span.enter();
+    let mut context_propagator = HashMap::<String, String>::default();
+    let context = span.context();
+    global::get_text_map_propagator(|propagator| {
+        propagator.inject_context(&context, &mut context_propagator);
+    });
```
Comment (vlad-ivanov-name) [inline josh-proxy/src/bin/josh-proxy.rs:636]: can this be wrapped into something with a meaningful name? like `propagate_span` or something

Comment (vlad-ivanov-name) [inline josh-proxy/src/bin/josh-proxy.rs:636]: does `global` mean that global program state is being manipulated here? would be helpful if there's a comment about that

Comment (LMG) [reply josh-proxy/src/bin/josh-proxy.rs:636]: Here we're only reading from the global configuration. This is commented on the writing side

Comment (LMG) [reply josh-proxy/src/bin/josh-proxy.rs:636]: sorry, not the writing side, but the part where I set it up. This is the part where we write in it :D

Comment (LMG) [reply josh-proxy/src/bin/josh-proxy.rs:636]: I am not sure it makes sense. From my understanding, span and _enter need to be declared in this context to link stuff properly. Maybe the three remaining lines then?

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**